### ISSUE TITLE
Something is up with hg19 in our installation

### ISSUE THREAD

**Issue by Wixaros** [chapmanb/bcbio-nextgen#2219]
Hi

I'm working on validating some results, and have received some bamfiles, in which they used hg19 as reference. Our current BCbio install only had GRch37 so I tried to install the hg19 with the following command:
`bcbio_nextgen.py upgrade --genomes hg19`

Looked like it worked, however when I try to run a test, I get the following error

`[2018-01-15T10:52Z] System YAML configuration: /usr/local/share/packages/bcbio/galaxy/bcbio_system.yaml
[2018-01-15T10:52Z] Resource requests: sambamba, samtools; memory: 5.00, 5.00; cores: 6, 6
[2018-01-15T10:52Z] Configuring 1 jobs to run, using 6 cores each with 30.1g of memory reserved for each job
[2018-01-15T10:52Z] Timing: organize samples
[2018-01-15T10:52Z] multiprocessing: organize_samples
[2018-01-15T10:52Z] Using input YAML configuration: /casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/config/test4.yaml
[2018-01-15T10:52Z] Checking sample YAML configuration: /casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/config/test4.yaml
[2018-01-15T10:52Z] Downloading hg19 samtools from AWS
Traceback (most recent call last):
  File "/usr/local/bin/bcbio_nextgen.py", line 234, in <module>
    main(**kwargs)
  File "/usr/local/bin/bcbio_nextgen.py", line 43, in main
    run_main(**kwargs)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 42, in run_main
    fc_dir, run_info_yaml)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 86, in _run_toplevel
    for xs in pipeline(config, run_info_yaml, parallel, dirs, samples):
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/main.py", line 119, in variant2pipeline
    [x[0]["description"] for x in samples]]])
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/multi.py", line 28, in run_parallel
    return run_multicore(fn, items, config, parallel=parallel)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/multi.py", line 86, in run_multicore
    for data in joblib.Parallel(parallel["num_jobs"], batch_size=1)(joblib.delayed(fn)(x) for x in items):
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 758, in __call__
    while self.dispatch_one_batch(iterator):
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 608, in dispatch_one_batch
    self._dispatch(tasks)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 571, in _dispatch
    job = self._backend.apply_async(batch, callback=cb)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/joblib/_parallel_backends.py", line 109, in apply_async
    result = ImmediateResult(func)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/joblib/_parallel_backends.py", line 326, in __init__
    self.results = batch()
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/joblib/parallel.py", line 131, in __call__
    return [func(*args, **kwargs) for func, args, kwargs in self.items]
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/utils.py", line 51, in wrapper
    return apply(f, *args, **kwargs)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/distributed/multitasks.py", line 316, in organize_samples
    return run_info.organize(*args)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/run_info.py", line 66, in organize
    item = add_reference_resources(item, remote_retriever)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/run_info.py", line 158, in add_reference_resources
    data["reference"] = genome.get_refs(data["genome_build"], aligner, data["dirs"]["galaxy"], data)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/genome.py", line 207, in get_refs
    galaxy_config, data)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/genome.py", line 159, in _get_ref_from_galaxy_loc
    cur_ref = download_prepped_genome(genome_build, data, name, need_remap)
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/pipeline/genome.py", line 280, in download_prepped_genome
    raise ValueError("Could not find reference genome file %s %s" % (genome_build, name))
ValueError: Could not find reference genome file hg19 samtools
`

I then tried to upgrade everything, using 
`bcbio_nextgen.py upgrade -u stable --tools --data --genomes hg19`

However this too stalled, when trying to connect to a ncbi-ftp:
`--2018-01-15 14:29:32--  ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv3.3.2/GRCh37//HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_noinconsistent.bed
  (try:14) => "-"
Connecting to ftp-trace.ncbi.nlm.nih.gov|130.14.250.13|:21... connected.
Logging in as anonymous ...
Error in server response, closing control connection.
Retrying.

--2018-01-15 14:29:43--  ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv3.3.2/GRCh37//HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_noinconsistent.bed
  (try:15) => "-"
Connecting to ftp-trace.ncbi.nlm.nih.gov|130.14.250.13|:21... connected.
Logging in as anonymous ...
Error in server response, closing control connection.
Retrying.

--2018-01-15 14:29:53--  ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv3.3.2/GRCh37//HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_noinconsistent.bed
  (try:16) => "-"
Connecting to ftp-trace.ncbi.nlm.nih.gov|130.14.250.13|:21... connected.
Logging in as anonymous ...
Error in server response, closing control connection.
Retrying.

--2018-01-15 14:30:03--  ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv3.3.2/GRCh37//HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_noinconsistent.bed
  (try:17) => "-"
Connecting to ftp-trace.ncbi.nlm.nih.gov|130.14.250.13|:21... connected.
Logging in as anonymous ...
Error in server response, closing control connection.
Retrying.

--2018-01-15 14:30:13--  ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv3.3.2/GRCh37//HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_noinconsistent.bed
  (try:18) => "-"
Connecting to ftp-trace.ncbi.nlm.nih.gov|130.14.250.13|:21... connected.
Logging in as anonymous ...
Error in server response, closing control connection.
Retrying.

--2018-01-15 14:30:23--  ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv3.3.2/GRCh37//HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_noinconsistent.bed
  (try:19) => "-"
Connecting to ftp-trace.ncbi.nlm.nih.gov|130.14.250.13|:21... connected.
Logging in as anonymous ...
Error in server response, closing control connection.
Retrying.

--2018-01-15 14:30:34--  ftp://ftp-trace.ncbi.nlm.nih.gov/giab/ftp/release/AshkenazimTrio/HG002_NA24385_son/NISTv3.3.2/GRCh37//HG002_GRCh37_GIAB_highconf_CG-IllFB-IllGATKHC-Ion-10X-SOLID_CHROM1-22_v.3.3.2_highconf_noinconsistent.bed
  (try:20) => "-"
Connecting to ftp-trace.ncbi.nlm.nih.gov|130.14.250.13|:21... connected.
Logging in as anonymous ...
Error in server response, closing control connection.
Giving up.

Traceback (most recent call last):
  File "/usr/local/bin/bcbio_nextgen.py", line 215, in <module>
    kwargs["fc_dir"] = fc_dir
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/install.py", line 95, in upgrade_bcbio
    print("Upgrading bcbio-nextgen data files")
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/site-packages/bcbio/install.py", line 290, in upgrade_bcbio_data
    sys.path.insert(0, cbl["dir"])
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/deploy/__init__.py", line 65, in deploy
    _setup_vm(options, vm_launcher, actions)
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/deploy/__init__.py", line 110, in _setup_vm
    configure_instance(options, actions)
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/deploy/__init__.py", line 268, in configure_instance
    setup_biodata(options)
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/deploy/__init__.py", line 250, in setup_biodata
    install_proc(options["genomes"], ["ggd", "s3", "raw"])
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/genomes.py", line 338, in install_data
    _prep_genomes(env, genomes, genome_indexes, ready_approaches)
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/genomes.py", line 467, in _prep_genomes
    retrieve_fn(env, manager, gid, idx)
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/genomes.py", line 823, in _install_with_ggd
    ggd.install_recipe(env.cwd, recipe_file, gid)
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/ggd.py", line 31, in install_recipe
    recipe["recipe"]["full"]["recipe_type"])
  File "/casa3/nobackup/med-nca/MaNiLa/GATC_MaNiLa_TP53_project/test4/work/tmpbcbio-install/cloudbiolinux/cloudbio/biodata/ggd.py", line 63, in _run_recipe
    subprocess.check_output(["bash", run_file])
  File "/usr/local/share/packages/bcbio/anaconda/lib/python2.7/subprocess.py", line 219, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['bash', '/state/partition1/apps/local/share/packages/bcbio/genomes/Hsapiens/hg19/txtmp/ggd-run.sh']' returned non-zero exit status 4
`

I can't seem to find the error, other than hg19 might not be fully accessible at some point, locally or online.

[test4.txt](https://github.com/chapmanb/bcbio-nextgen/files/1631815/test4.txt)
Attached is the .yaml configuration that I'm running the test with.

**Comment by chapmanb** [comment]
Thanks much for the detailed report and apologies about the issues. It looks like something may have been problematic about your initial install and hg19 never got fully installed. You should be able to check:
```
/state/partition1/apps/local/share/packages/bcbio/galaxy/tool-data/sam_fa_indices.loc
```
to see if it has an hg19 entry pointing to an existing fasta file on your system. If not, something may have gone wrong.

For the re-upgrade where you're stuck on the download, do you see the same problem if you re-run multiple times? It looks from here like the NCBI ftp server is up and access to that file is fine. I'm not positive if you were experiencing a temporary downtime on their side, or something is problematic about your connection to their server. If you restart with:
```
bcbio_nextgen.py upgrade --data --genomes hg19
```
it should pick up where it left off and hopefully work cleanly this time. Hope this helps some with getting it installed and working.

**Comment by Wixaros** [comment]
Hi again

Sorry for the long response time. We have figured out a lot of stuff in the meantime and have now gotten to the place where we have a problem with BCbio not recognizing the locations of different files. I have located the error (just like the above error), however is there a way for us to modify the path with a default directory?
The problem is that we are running the script on a Sun Grid Engine through a `qsub` command and therefore the installing directories are not quite the same as the filepaths from the actual cluster. So all the filepaths should be modified just bit.

from
`/state/partition1/apps/local/share/packages/bcbio/etc`
to
`/usr/state/partition1/apps/local/share/packages/bcbio/etc`

Thanks for a really nice job!
